src/authenticator.ts

```diff
@@ -63,7 +70,7 @@ export default class Authenticator {
     };
   }
 
-  private authenticateWithRefreshToken(oldRefreshToken: string, options: AuthenticateOptions): AuthenticationResponse {
+  private authenticateWithRefreshToken(oldRefreshToken: string, options: AuthenticateOptions): AuthenticationResponse | ErrorResponse {
```

Comment:
I think the change to using `ErrorResponse` instead of specialised exceptions is good, but I'm not sure about using the return type for that. I don't like exceptions too much either, to be honest. Maybe we should unify the success and error cases, if we expect users to just forward them to the client. WDYT?

Reply:
Potentially, but unify them how? Isn't specifying the return type as a union type, unification?